packages/babel-cli/src/babel/dir.js

```diff
@@ -138,24 +139,50 @@ export default async function({
     }
   }
 
+  let compiledFiles = 0;
+  let startTime = null;
+
+  const logSuccess = debounce(
+    function() {
+      if (startTime === null) {
+        // This should never happen, but just in case it's better
+        // to ignore the log message rather than making @babel/cli crash.
+        return;
+      }
+
+      const diff = process.hrtime(startTime);
+
+      console.log(
+        `Successfully compiled ${compiledFiles} ${
+          compiledFiles !== 1 ? "files" : "file"
+        } with Babel (${diff[0] * 1e3 + Math.round(diff[1] / 1e6)}ms).`,
+      );
+      compiledFiles = 0;
```

Comment:
I think we should also reset `compiledFiles` if any of
- `cliOptions.quiet` is `true`.
- error is thrown in `await handle(filename)`.

Reply:
> cliOptions.quiet is true.

- Why? If `quiet` is true, we never read this variable so it doesn't really matter.

> error is thrown in await handle(filename).

- In this case, the `compiledFiles` variable would just be thrown away because it's defined in the same scope as the one containing `await handle(filename)`.

Reply:
If `babel` is running on watch mode, any errors in the first build will not reset `compiledFiles` nor halt the execution. The `compiledFiles` will be accumulated until `logSuccess()` is called. It may confuse users if they revise a single file to fix the syntax error and `babel` prints `Successfully compiled 2366 files`.

Reply:
Oh my bad! I thought `handle(filename)` will be returned for some unknown mental process. You are right!